packages/plugins/src/plugins/eigenlayer/fetchers/withdrawal.ts

```diff
@@ -0,0 +1,79 @@
+import {
+  NetworkId,
+  PortfolioElement,
+  PortfolioElementType,
+} from '@sonarwatch/portfolio-core';
+import { Cache } from '../../../Cache';
+import { Position, Withdrawal } from '../types';
+import { platformId } from '../constants';
+import { getAddress } from 'viem';
+
+import { getAssetsFromPositions } from '../helper';
+
+/**
+ * Returns the withdrawals for a given owner
+ * @param owner - The address of the owner
+ * @param cache - The cache instance
+ * @returns The withdrawals
+ */
+export const getWithdrawals = async (owner: string, cache: Cache) => {
+  const withdrawals = await cache.getItem<Withdrawal[]>(
+    'eigenlayer-withdrawals',
+    {
+      prefix: platformId,
+      networkId: NetworkId.ethereum,
+    }
+  );
+
+  const strategies = await cache.getItem<Position[]>('eigenlayer-strategies', {
+    prefix: platformId,
+    networkId: NetworkId.ethereum,
+  });
+
+  const userWithdrawals = withdrawals?.filter(
+    (withdrawal) =>
+      getAddress(withdrawal.withdrawerAddress) === getAddress(owner)
+  );
+
+  const shares = userWithdrawals?.flatMap((withdrawal) => withdrawal.shares);
+
+  if (!shares || shares.length === 0) return [];
+
+  // Build the positions
+  const positions: Position[] = shares.map((share) => ({
+    strategyAddress: share.strategyAddress,
+    amount: share.shares,
+    underlyingToken: strategies?.find(
+      (strategy) =>
+        getAddress(strategy.strategyAddress) ===
+        getAddress(share.strategyAddress)
+    )?.underlyingToken,
+    decimals: strategies?.find(
+      (strategy) =>
+        getAddress(strategy.strategyAddress) ===
+        getAddress(share.strategyAddress)
+    )?.decimals,
```

Comment:
Can we post-process _once_ the result we get from the cache for `withdrawals` and `strategies` and stop constantly calling `getAddress()` ?

Wouldn't this simplify/lighten the code and allow us to do something like:
```suggestion
       strategy.strategyAddress ===
       share.strategyAddress
    )?.underlyingToken,
    decimals: strategies?.find(
      (strategy) =>
       strategy.strategyAddress ===
       share.strategyAddress
    )?.decimals,
```

Or, as an alternative, build small lookup tables that contain all the `underlyingToken` and `decimals` values (using `strategy` as the keys) to avoid constantly calling `find()` on the same data set ?